model/metric_test.go

```diff
@@ -269,6 +277,14 @@ func TestEscapeName(t *testing.T) {
 			expectedUnescapedDots: "http.status:sum",
 			expectedValue:         "U__http_2e_status:sum",
 		},
+		{
+			name:                  "name with spaces and emoji",
+			input:                 "label with 😱",
+			expectedUnderscores:   "label_with__",
+			expectedDots:          "label_with__",
+			expectedUnescapedDots: "label_with_",
```

Comment:
I'm confused a bit by this -- `expectedUnescapedDots` is the expected output of un-escpaing `input` with the `DotsEscaping`? What if the input was not a valid escape sequence, like this case? Wouldn't we want to error out instead?

Reply:
the input is valid, and all non-legacy characters become underscores. We never want to produce an error.

However it's true that two non-legacy characters in a row become a double underscore, which the unescaper translates as a single underscore.  I am not sure there is a better alternative here, though. (non-legacy characters become `__`?)

Reply:
I got confused with the test, sorry. I see that `expectedUnescapedDots` is the expected unescaping of the already escaped input. Looks good.